### DISCUSSION TITLE
immich isnt safe for store your photos and videos ?

### DISCUSSION THREAD

**Discussion by masterwishx** [immich-app/immich#7036] [Q&A] [answered]
i want to check this app , currently using Photoprism , but found:   
` Do not use the app as the only way to store your photos and videos.` 

is this mean that it isn't safe to use this app at moment or when it will be safe ? 
how to understand this ? files can be deleted somehow ?
Thanks

**Comment by schuhbacca** [comment]
It means immich is still in very active development. Obviously there's never an intention to lose data but it's still under modification and things happen. As with any data you don't want to lose always have a good backup strategy.

**Reply by masterwishx** [reply]
Sure, but sometime a lot of photos and videos take much space, so a little hard to save another xTB of data in other space...

**Reply by schuhbacca** [reply]
Valid. All what you are willing to stomach in the end. Self hosting isn't cheap. Even if immich worked perfectly multiple backups of data are recommended. But for now immich is still non stable and data protection is highly encouraged by the devs.

**Reply by arkus7** [reply]
I have recently started using Immich. My idea for backups is like this:
- Store some of the photos on the hard drive (I don't have so much space, that's why some)
- Have all photos uploaded to Immich
- Keep a backup of photos uploaded to Immich on cloud storage, such as Amazon S3 or Backblaze B2

Since Immich doesn't allow us to connect to S3 compatible storage directly yet (#4445), I store them on the server disk with backups enabled. I plan to add the 3rd part (syncing them to object storage) soon, to keep the library backed up in another place, just for safety.

Obviously, if you have TBs of data, the disk cost might be big. I recently started using [Hetzner's StorageBox](https://www.hetzner.com/storage/storage-box/), which is way cheaper than regular disks attached to the VPS and so far have no issues.

**Reply by masterwishx** [reply]
I'm using for now selfhosted PhotoPrism in home sever in Unraid. But very interested in immich.
So about 100gb imported and 400gb still not imported. And if I will add famaly it will grow to xTB.
In Unraid I have parity disk but it's not backup. Also have Oracle free vps server but 200gb.
So for pictures/video backup maybe zfs snapshot good idea? Needs to be checked...
Or just to wait when developers remove this warning if it possible?

**Reply by bo0tzz** [reply, marked as answer]
Warning or not, you should have proper backups on a remote system. ZFS snapshots are not a backup.

**Reply by masterwishx** [reply]
Thanks got it

**Comment by Thyrfing89** [comment]
Question about this. 

Would you all recommand to backup Photos to another location before adding to Immich? Or would a control that Photos uploaded to Immich and then take a backup of the Immich Library and database be enough?

**Reply by schuhbacca** [reply]
Personally I back them up to immich then back up that image library. Immichs Golden rule is it doesn't touch the original files so it's safe enough.

**Reply by Thyrfing89** [reply]
> Personally I back them up to immich then back up that image library. Immichs Golden rule is it doesn't touch the original files so it's safe enough. 

Thank you, this is the route i wanna go myself. Library is on a Raid1 Nas, and will be backed up to another raid1 nas, Plus external harddrives:)

**Comment by Desarso** [comment]
I have a raid duplication set up with 8tb, run a cron job that checks if the raid array is working every hour

**Reply by Thyrfing89** [reply]
Awesome idea!
